src/coreclr/src/.nuget/Microsoft.NET.Sdk.IL/targets/Microsoft.NET.Sdk.IL.targets

```diff
@@ -144,6 +144,39 @@ Copyright (c) .NET Foundation. All rights reserved.
     <CallTarget Targets="$(TargetsTriggeredByCompilation)" Condition="'$(TargetsTriggeredByCompilation)' != ''"/>
   </Target>
 
+  <Target Name="GenerateVersionFile">
+   <PropertyGroup>
+      <VersionFilePath Condition="'$(VersionFilePath)' == ''">$([MSBuild]::NormalizePath('$(IntermediateOutputPath)', 'version', 'version.h'))</VersionFilePath>
+      <_AssemblyVersion>$(AssemblyVersion.Replace('.', ':'))</_AssemblyVersion>
+      <_CoreAssemblyVersion>$(CoreAssemblyVersion.Replace('.', ':'))</_CoreAssemblyVersion>
+      <_VersionFileContents>
+<![CDATA[
+#define CORE_ASSEMBLY "$(CoreAssembly)"
+#define ASSEMBLY_VERSION "$(_AssemblyVersion)"
+#define CORE_ASSEMBLY_Version "$(_CoreAssemblyVersion)"
+$(ExtraMacros)
+
+// Metadata version: v4.0.30319
+.assembly extern CORE_ASSEMBLY
+{
+  .publickeytoken = ($(_PublicKeyToken) )
```

Comment:
Skip emitting this when the `_PublicKeyToken` is empty? 

Also, should `_PublicKeyToken` be without underscore prefix? I believe that underscore prefix means implementation detail of the target that is not meant to be set from outside.